airflow/assets/__init__.py

```diff
@@ -378,12 +386,18 @@ class Dataset(Asset):
 
     asset_type: ClassVar[str] = "dataset"
 
+    def __init__(self, *args, group="dataset", **kwargs) -> None:
+        super().__init__(*args, group=group, **kwargs)
```

Comment:
I think we should make this smarter in the base class so subclasses’ `group` default to `asset_type` automatically. Requiring all subclasses to implement `__init__` like this is problematic.